onnxruntime/core/providers/dnnl/subgraph/dnnl_conv.h

```diff
@@ -470,29 +470,29 @@ class DnnlConv : public DnnlKernel {
     std::string auto_pad;
     auto attr = attributes.find(attributes_prefix + "auto_pad");
     if (attr != attributes.end() &&
-        attr->second->type() == ::ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_STRING) {
-      auto_pad = attr->second->s();
+        attr->second().type() == ::ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_STRING) {
```

Comment:
> attr->second().type() == ::ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_STRING [](start = 7, length = 109)

tensorprotoutils.h has a lot of helpers to help migration to proto3. I think those need to be adjusted. For this case I think there is utils::IsString

Reply:
Hmm, but those are internal to the runtime code and providers don't have access to them currently.